omegatrade-app/frontend/src/app/components/simulation/simulation.component.ts

```diff
@@ -0,0 +1,206 @@
+                  this.snackBarService.openSnackBar(response.message, '');
+                  const index = this.dataSource.data.findIndex(simulationObj => simulationObj.sId === simulation.sId);
+                  if (index > -1) {
+                    this.dataSource.data.splice(index, 1)
+                    this.dataSource = new MatTableDataSource(this.dataSource.data);
+                    this.runningSimulation = this.dataSource.data.length;
+                    this.initializeSortAndPagination();
+                  }
+                  this.updateCompanyStatus(simulation.companyId);
+                }
+                this.loader = false;
+              }, error => {
+                if (error && error.error && error.error.message) {
+                  this.snackBarService.openSnackBar(error.error.message, '');
+                }
+                this.loader = false;
+              }
+            );
+        }
+      });
+    }
+  }
+
+  simulate(formDirective: FormGroupDirective) {
+    if(this.runningSimulation >= this.maxAllowedSimulation){
+      this.snackBarService.openSnackBar(`Cannot simulate more than ${this.maxAllowedSimulation} simulations`, '');
+      return false;
+    }
+    this.loader = true;
+    this.restService.postData('simulations/start', this.simulateForm.value)
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.updateCompanyStatus(this.simulateForm.get('companyId').value, true);
+            this.simulateForm.reset();
+            formDirective.resetForm();
+            this.getSimulations();
+          }
+          this.loader = false;
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.loader = false;
+        });
+  }
+
+  getSimulations() {
+    this.loader = true;
+    this.restService.getData('simulations/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.runningSimulation = response.data.length;
+            console.log(this.runningSimulation);
+            this.dataSource = new MatTableDataSource(response.data);
+            this.initializeSortAndPagination();
+            if (response.data && response.data.length > 0) {
+              for (var i = 0; i < response.data.length; i++) {
+                this.updateCompanyStatus(response.data[i].companyId, true)
+              }
+            }
+          }
+          this.loader = false;
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.loader = false;
+        });
+  }
+
+  updateSimulation(simulation) {
+    this.loader = true;
+    const payLoad = { sId: simulation.sId, status: (simulation.status) ? false : true }
```

Comment:
What is going on here: `status: (simulation.status) ? false : true` ? Is it flipping the status from true to false every time it is updated? What does the status mean?

Reply:
simulation.status is old status from DB
for example:
if user wants to stop a simulation from running state, this function will change the status and update in DB.

But this commit is old commit, This `status` field/column is later changed to `STRING`. it will have only three status
`PROCESSING`,`STOPPED`,`COMPLETED`